### PR TITLE
Handle storage types in staticresourcecontroller

### PR DESCRIPTION
**Pull request by bertinatto**
CC @openshift/storage

### CODE REVIEW

**Comment by bertinatto** [comment]
/assign @deads2k 
David, could you take a look please?

File: pkg/operator/resource/resourceapply/storage.go [hunk 1]
```diff
@@ -85,3 +85,33 @@ func ApplyCSIDriverV1Beta1(client storageclientv1beta1.CSIDriversGetter, recorde
...
+	if !*modified {
+		return existingCopy, false, nil
+	}
+
+	if klog.V(4) {
+		klog.Infof("CSIDriver %q changes: %v", required.Name, JSONPatchNoError(existing, existingCopy))
+	}
+
+	// TODO: Spec is read-only, so this will fail if user changes it. Should we simply ignore it?
+	actual, err := client.CSIDrivers().Update(context.TODO(), existingCopy, metav1.UpdateOptions{})
```
Comment (sttts) [inline pkg/operator/resource/resourceapply/storage.go:114]: I think this is fine.

File: pkg/operator/resource/resourceapply/generic.go [hunk 2]
```diff
@@ -154,6 +155,16 @@ func ApplyDirectly(clients *ClientHolder, recorder events.Recorder, manifests As
 				result.Error = fmt.Errorf("missing apiExtensionsClient")
 			}
 			result.Result, result.Changed, result.Error = ApplyCustomResourceDefinitionV1(clients.apiExtensionsClient.ApiextensionsV1(), recorder, t)
+		case *storagev1.StorageClass:
+			if clients.kubeClient == nil {
+				result.Error = fmt.Errorf("missing kubeClient")
+			}
+			result.Result, result.Changed, result.Error = ApplyStorageClass(clients.kubeClient.StorageV1(), recorder, t)
```
Comment (sttts) [inline pkg/operator/resource/resourceapply/generic.go:162]: where is ApplyStorageClass? Missing file?

Comment (bertinatto) [reply pkg/operator/resource/resourceapply/generic.go:162]: It already exists, here:

https://github.com/openshift/library-go/pull/815/files/54705a0472e861a45c437f0f8592a0e78dca37ab#diff-3de2a0c77901abef949759ee8869c9d1L21

Comment (sttts) [reply pkg/operator/resource/resourceapply/generic.go:162]: ic

**Pull request merged**